src/app/messages/page.tsx

```diff
@@ -1,30 +1,64 @@
-import { createServerComponentClient } from '@supabase/auth-helpers-nextjs'
+import MessagePreview from '@/components/MessagePreview'
+import Button from '@/components/common/Button'
+import Link from 'next/link'
+import Image from 'next/image'
 import { redirect } from 'next/navigation'
 import { Database } from '@/types/supabase'
 import { cookies } from 'next/headers'
+import { createServerComponentClient } from '@supabase/auth-helpers-nextjs'
+import { css } from '../../../styled-system/css'
 
 export default async function MessagesPage() {
   const supabase = createServerComponentClient<Database>({ cookies })
 
-  const { data } = await supabase.auth.getSession()
+  const {
+    data: { session },
+  } = await supabase.auth.getSession()
 
-  if (!data.session) {
+  if (!session) {
     return redirect('/login')
   }
 
-  const { data: contacts } = await supabase
-    .from('contacts')
-    .select('contact_id, user_id, contact_user_id')
-    .eq('user_id', data.session.user.id)
+  const { data: contacts } = await supabase.from('profiles').select('*')
+
+  // Note: I think there is no way to query Supabase for all chats that a user is in since users is an array type.
```

Comment:
Interesting limitation. Have you asked in their community (github, slack etc) directly? I'm curious to know their official take on it.